### PR TITLE
fix: disable timeout at RoundTripper level & improve logging

### PR DESCRIPTION
**Pull request by MatousJobanek**
This PR fixes two things. Why two in the same PR? because they are related :stuck_out_tongue_winking_eye: 

First of all, the PR changes the way we define values for the RoundTripper. The `http.DefaultTransport` sets 30s timeout as part of the DialerContext by default, so we need to change that  and set it to zero. 

The second fix is related to logging. The original log (produced as part of the `middleware.RequestLoggerWithConfig`) was printed out only when the routing was finished, so we didn't see these watch calls in the logs. The reason is that they are just a long-running GET calls that open the stream and the routing is not really finished. Thus, I added another log that prints out when the request is received. As part of this, I also modified the format so we use standard logger and not `fmt.Printf`. There is still a room for an improvement of the standard logs as well - see my TODO comments - but let's do it in the next PR.

it's the second (and last step) to fix  https://issues.redhat.com/browse/SANDBOX-85

paired with: https://github.com/codeready-toolchain/toolchain-e2e/pull/753

### CODE REVIEW

**Comment by MatousJobanek** [comment]
cc @jhutar

**Comment by jhutar** [comment]
Perf team is looking forward to have this in RHTAP Stage :-)

**Review by alexeykazakov — APPROVED** [review]
Nice catch!

**Review by rajivnathan — APPROVED** [review]
Very nice! The log changes will play nicer with logging tools indeed. 👍 I should have thought of that 🤦 . Just added 1 question but it's very minor so shouldn't block merging whenever you're ready.

File: pkg/proxy/proxy.go [hunk 1]
```diff
@@ -349,25 +356,47 @@ func (p *Proxy) newReverseProxy(ctx echo.Context, target *access.ClusterAccess,
...
+func noTimeoutDefaultTransport() *http.Transport {
+	return &http.Transport{
+		Proxy:                 http.ProxyFromEnvironment,
+		DialContext:           noTimeoutDialerProxy,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply pkg/proxy/proxy.go]: funny, I did exactly that as the first approach, but I changed it several times when working on unit tests. The original way I wanted to test it didn't work anyway and I somehow got to this version. 
But having it as it is - yes, you are right, I can easily replace it with a copy of the DefaultTransport e7279f3

**Comment by MatousJobanek** [comment]
/retest
flaky cleanup https://storage.googleapis.com/origin-ci-test/pr-logs/pull/codeready-toolchain_registration-service/332/pull-ci-codeready-toolchain-registration-service-master-e2e/1672206369115082752/build-log.txt

File: pkg/proxy/proxy.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rajivnathan) [reply pkg/proxy/proxy.go]: Nice 👍

**Review by xcoulon — APPROVED** [review]

**Pull request merged**